### PR TITLE
[Workspace] AG-Grid disabled

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Description

Please include a summary of the change and which issue is fixed.
List any external dependencies that are required for this change.

Completes: AB#FILL_IN_YOUR_ISSUE_ID

## Checklist:

-   [ ] I have performed a self-review of my own code.
-   [ ] I have linked my DevOps task using the AB# tag.
-   [ ] My code is easy to read, and comments are added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: src/Core/WorkSpace/src/Tabs/GridTab.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import { AgGridReact } from 'ag-grid-react';
+import { WorkspaceFilter } from '../Components/WorkspaceFilter/WorkspaceFilter';
+import { useMemo, useState } from 'react';
+import { useFilterApiContext } from '@equinor/filter';
+import { useWorkSpace } from '@equinor/WorkSpace';
+import { useDataContext } from '../Context/DataProvider';
+import { QueryClient, useQueryClient } from 'react-query';
+import { TableConfigBar } from '../../../../packages/Table/Components/TableConfigBar/TableConfigBar';
+import { agStyles } from '../../../../packages/AgGrid/styles/styles';
+import useAgGridStyles from '../../../../packages/AgGrid/hooks/useAgGridStyles';
```
Comment (Noggling) [inline src/Core/WorkSpace/src/Tabs/GridTab.tsx:11]: Alias

File: src/packages/Filter/Hooks/useFilterApi.ts [hunk 2]
```diff
@@ -358,6 +372,23 @@ export function useFilterApi<T>({
         handleShouldReRender(preventReRender);
     }
 
+    function removeOnFilterChanged(id: string) {
+        console.log('removing subscriber from filter');
```
Comment (Gustav-Eikaas) [inline src/packages/Filter/Hooks/useFilterApi.ts:376]: remove

File: src/packages/Filter/Hooks/useFilterApi.ts [hunk 3]
```diff
@@ -358,6 +372,23 @@ export function useFilterApi<T>({
...
+        onFilterChangedCallbacks.current = onFilterChangedCallbacks.current.filter(
+            (s) => s.id !== id
+        );
+    }
+
+    function registerOnFilterChangedCallback(callback: () => void) {
+        const uniqueId = (Math.random() * 16).toString();
```
Comment (Gustav-Eikaas) [inline src/packages/Filter/Hooks/useFilterApi.ts:383]: remove

File: src/Core/WorkSpace/src/WorkSpaceApi/workspaceApi.ts [hunk 4]
```diff
@@ -185,7 +186,62 @@ export function createWorkSpace<T>(options: ViewerOptions<T>): WorkSpaceApi<T> {
 
             return workspaceAPI;
         },
+        // registerGridOptions(gridOptions: GridConfig<T>) {
```
Comment (Noggling) [inline src/Core/WorkSpace/src/WorkSpaceApi/workspaceApi.ts:189]: uncomment ;)

File: src/Core/WorkSpace/src/Tabs/GridTab.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+const useGridContext = (): GridContext => {
+    const queryClient = useQueryClient();
+    const { name } = useWorkSpace();
+
+    return {
+        queryClient,
+        refetchData: () => queryClient.invalidateQueries(name),
+        workspaceName: name,
+    };
+};
```
Comment (Gustav-Eikaas) [inline src/Core/WorkSpace/src/Tabs/GridTab.tsx:105]: Extract

File: src/Core/WorkSpace/src/WorkSpaceApi/WorkSpaceTypes.ts [hunk 6]
```diff
@@ -74,6 +75,7 @@ export interface WorkSpaceApi<T> {
     registerPowerBIOptions: (options: PowerBiOptions) => WorkSpaceApi<T>;
     registerWorkflowEditorOptions: (options: WorkflowEditorOptions) => WorkSpaceApi<T>;
     registerPresets: (options: PresetOption[]) => WorkSpaceApi<T>;
+    // registerGridOptions: (gridOptions: GridConfig<T>) => WorkSpaceApi<T>;
```
Comment (Gustav-Eikaas) [inline src/Core/WorkSpace/src/WorkSpaceApi/WorkSpaceTypes.ts:78]: unComment

File: src/Core/WorkSpace/src/WorkSpaceApi/workspaceApi.ts [hunk 7]
```diff
@@ -185,7 +186,62 @@ export function createWorkSpace<T>(options: ViewerOptions<T>): WorkSpaceApi<T> {
...
+                onCellClicked: (ev) => ev.node.setSelected(true, true),
+                checkboxSelection: true,
+            },
+            ...processedGridOptions,
+        ],
+    };
+}
+//Handles checkbox selection
+const handleSelectionChangedEvent = <T>(
+    e: SelectionChangedEvent,
```
Comment (Gustav-Eikaas) [inline src/Core/WorkSpace/src/WorkSpaceApi/workspaceApi.ts:245]: Extract

File: src/Core/WorkSpace/src/Tabs/GridTab.tsx [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+    );
+};
+
+interface WorkspaceGridProps {
+    gridOptions: GridOptions;
+    onGridReady?: (gridApi: GridApi, columnApi: ColumnApi) => void;
+    context?: GridContext;
+}
+
+export const WorkspaceGrid = ({
```
Comment (Noggling) [inline src/Core/WorkSpace/src/Tabs/GridTab.tsx:58]: Ut i egen fil / pakke

File: src/Core/WorkSpace/src/WorkSpaceApi/workspaceApi.ts [hunk 9]
```diff
@@ -185,7 +186,62 @@ export function createWorkSpace<T>(options: ViewerOptions<T>): WorkSpaceApi<T> {
...
+    const processedGridOptions = gridOptions.columns.map(
+        (opt): ColDef => ({
+            onCellClicked: opt.onClickOpensSidesheet
+                ? (ev: CellClickedEvent) => onSelect(ev.data)
+                : undefined,
+            resizable: true,
+            sortable: true,
+            cellStyle: opt.onClickOpensSidesheet ? { cursor: 'pointer' } : undefined,
+            enableRowGroup: true,
+            headerName: opt.title,
```
Comment (Noggling) [inline src/Core/WorkSpace/src/WorkSpaceApi/workspaceApi.ts:223]: headerName ikke title

**Pull request closed without merging**